e2e/sample-apps/compat.js

```diff
@@ -0,0 +1,270 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import firebase from 'firebase/compat/app';
+import 'firebase/compat/auth';
+import 'firebase/compat/analytics';
+import 'firebase/compat/app-check';
+import 'firebase/compat/functions';
+import 'firebase/compat/storage';
+import 'firebase/compat/firestore';
+import 'firebase/compat/messaging';
+import 'firebase/compat/performance';
+import 'firebase/compat/database';
+/**
+ * The config file should look like:
+ *
+ * // A config for a project
+ * export const config = {
+ *   apiKey: ************,
+ *   authDomain: ************,
+ *   databaseURL: ************,
+ *   projectId: ************,
+ *   storageBucket: ************,
+ *   messagingSenderId: ************,
+ *   appId: ************,
+ *   measurementId: ************
+ * };
+ *
+ * // A user account with read/write privileges in that project
+ * // for storage, database, firestore
+ * export const testAccount = {
+ *   email: ************,
+ *   password: ************
+ * }
+ */
+import { config, testAccount } from '../firebase-config';
+
+/**
+ * Quick sample app to debug and explore basic Firebase API usage/problems.
+ *
+ * In order for app check to work, add the app check debug token to
+ * build/index.html and uncomment the line.
+ */
+
+/**
+ * Auth smoke test.
+ *
+ * Login with email and password. Account must exist. Should set up
+ * test project rules to only allow read/writes from this account
+ * (and other test accounts), to properly test rules.
+ *
+ * Logout after all tests are done.
+ */
+async function authLogin() {
+  const cred = await firebase
+    .auth()
+    .signInWithEmailAndPassword(testAccount.email, testAccount.password);
+  console.log('[AUTH] Logged in with test account', cred.user.email);
+  return cred;
+}
+async function authLogout() {
+  console.log('[AUTH] Logging out user');
+  return firebase.auth().signOut();
+}
+
+/**
+ * Functions smoke test.
+ *
+ * Call a deployed function.
+ * This cloud function must be deployed in this project first. It can be
+ * found in this repo's /functions folder.
+ */
+async function callFunctions() {
+  console.log('[FUNCTIONS] start');
+  const functions = firebase.functions();
+  const callTest = functions.httpsCallable('callTest');
```

Comment:
Do you need to run `yarn test:setup` at the root of the repo to make sure the function is deployed? I don't see it being called anywhere.

Reply:
I don't want to add too much to this project and I don't want to put this function in config/functions to keep this separate from main repo testing. I put the source code for the function itself in sample-apps/functions/index.js and it's up to the user to deploy it to their project. It's already deployed to the jscore-sandbox project.

Changed the comment to explain this.

Reply:
We should have a script to automate the deployment, so it's easy to set up again in case something changes in jscore-sandbox. Ideally we should always run this script as a setup step before running the tests without assuming the functions are available.

Reply:
Done. We don't do this for normal CI tests by the way. We might want to but it might be a little more complicated since the top level test setup script has interactive prompts and uses the firebase-cli module which seems to be a little buggy. Also the top level script also deploys database and firestore rules.

Reply:
Ok, I added a README.md explaining setup.

Reply:
I'm good as long as we have an easy way to deploy the function.